### ISSUE TITLE
Windows Touch Keyboard support 

### ISSUE THREAD

**Issue by CodeCharm** [microsoft/vscode#86996]
Issue Type: <b>Bug</b>

I know Code is multiplatform, and there may be limitations due to Electron, but using the Windows Touch Keyboard is truly a terrible experience, and sometimes I really need to or want to use the Touch Keyboard when I can't or don't want to use the physical keyboard

VS Code version: Code 1.41.0 (9579eda04fdb3a9bba2750f15193e5fafe16b959, 2019-12-11T18:37:42.077Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7660U CPU @ 2.50GHz (4 x 2496)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.93GB (8.44GB free)|
|Process Argv|C:\Users\alanm\OneDrive\Documents\PowerShell\profile.ps1|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (72)</summary>

Extension|Author (truncated)|Version
---|---|---
html-snippets|abu|0.2.1
vscode-json|and|1.5.0
vscode-color|ans|0.4.5
unique-lines|bib|1.0.0
LinkCheckMD|bla|0.1.5
vscode-toggle-quotes|Bri|0.3.2
npm-intellisense|chr|1.3.0
regex|chr|0.2.0
bracket-pair-colorizer-2|Coe|0.0.29
vscode-markdownlint|Dav|0.32.0
jsonpath-extract|dav|1.2.3
vscode-jq-playground|dav|2.1.1
jshint|dba|0.10.21
vscode-eslint|dba|1.9.1
docs-article-templates|doc|0.2.7
docs-authoring-pack|doc|0.1.3
docs-markdown|doc|0.2.27
docs-metadata|doc|1.0.4
docs-preview|doc|0.3.35
docs-yaml|doc|0.2.4
githistory|don|0.4.11
xml|Dot|2.5.0
gitlens|eam|10.2.0
EditorConfig|Edi|0.14.3
tslint|eg2|1.0.44
vscode-npm-script|eg2|0.3.9
vscode-npm|fkn|3.3.0
json2ts|Gre|0.0.6
live-html-previewer|hdg|0.3.0
vscode-guid|hea|1.4.19
beautify|Hoo|1.5.0
rest-client|hum|0.23.0
align-by-regex|jan|1.0.4
docthis|joe|0.7.1
MagicPython|mag|1.1.0
azure-pipelines|ms-|1.157.4
vscode-apimanagement|ms-|0.1.1
vscode-docker|ms-|0.9.0
mssql|ms-|1.7.1
python|ms-|2019.11.50794
remote-wsl|ms-|0.41.5
azure-account|ms-|0.8.7
cpptools|ms-|0.26.2
csharp|ms-|1.21.8
powershell|ms-|2019.11.0
vs-keybindings|ms-|0.2.0
vsliveshare|ms-|1.0.1377
vsonline|ms-|1.0.1370
team|ms-|1.161.0
debugger-for-chrome|msj|4.12.3
debugger-for-edge|msj|1.0.13
smartpaste|reb|0.0.3
vscode-yaml|red|0.6.1
vscode-statusbar-json-path|ric|1.4.0
partial-diff|ryu|1.4.0
preview-vscode|sea|2.0.3
code-settings-sync|Sha|3.4.3
vba|spe|1.2.0
vba-snippets|spe|1.0.1
vscode-vba-icons|spe|1.0.2
vscode-odata|sta|0.1.0
code-spell-checker|str|1.7.20
vscode-gulp|tan|0.0.4
pdf|tom|0.5.1
cmake|twx|0.0.17
lorem-ipsum|Tyr|1.2.0
vscodeintellicode|Vis|1.2.0
vscode-svg-previewer|vit|0.5.1
vscode-icons|vsc|9.6.0
bootstrap-3-snippets|wcw|0.1.0
JavaScriptSnippets|xab|1.7.2
html-css-class-completion|Zig|1.19.0

</details>

**Comment by RMacfarlane** [comment]
@CodeCharm Can you provide some more specifics about what doesn't work or what needs to be improved?

**Comment by AlanMcBee** [comment]
(I have two accounts [@AlanMcBee and @CodeCharm], sorry if it gets confusing)

Where to start? I have a Surface Pro (5th gen), but I'm sure I can repro this on my Surface Book 2 or any other touch-enabled Windows 10 1909 device.

This morning I tried to use the Touch Keyboard to use VSCode so that I could type silently. It was a miserable experience So much did not work well. The virtual trackpad was helpful, but half of my screen was devoted to input. It's super easy to see if you try to edit (not just type) anything using the Touch Keyboard. I don't think you need to be in Tablet Mode. I understand this is not the 80% use case for VSCode (I would not normally expect to be using only the Touch Keyboard to work with VSCode).

I does make me wonder whether VSCode is accessible for differently-abled people.

Anyway. A specific thing that does not work that would certainly help:

In most (admittedly not all) Windows apps and applications, as the cursor goes into the screen area that is obscured by the Touch Keyboard, Windows will try to scroll the app within the "viewport" area (the visible area in the client screen that is not dedicated to the keyboard) so that the area containing the cursor will be visible in the viewport.

To repro:
Using a touch-enabled Windows device
(optional) Put the device into Tablet mode (this helps because it makes sure VSCode stays maximized and might let Windows help move things around)
Launch VSCode with --disable-extensions
Display the Touch Keyboard (may need to enable the "Show the touch keyboard button" selection in the Notification Area, and then tap the Touch Keyboard button to display it.)
Make sure the Touch Keyboard is in Docked mode and not Floating mode
Using only the Touch Keyboard, press Ctrl+` to put the cursor into the Terminal view
Type things

Expected: Windows will shift VSCode up so that the cursor and the things you type are visible

Actual: You can type, but you can't see what is happening under the Touch Keyboard until you dismiss it

I wish Shift-Select would work from the Touch Keyboard, but I think that may be a limitation of Windows (I can't get it to work with Notepad or OneNote, either).

This is probably closely related to, but not a duplicate of #64925.

**Comment by RMacfarlane** [comment]
@AlanMcBee Thanks for taking the time to write this up!

**Comment by alexdima** [comment]
I will try to summarize the provided steps in https://github.com/microsoft/vscode/issues/86996#issuecomment-565759112

1. on a Windows Touch enabled device, enter Tablet mode
2. show the touch keyboard (using button if it does not happen automatically)
3. (make sure the Touch Keyboard is in Docked mode and not Floating mode)
4. the VS Code window is rendered "under" the touch keyboard.

@bpasero assigning you, as this is a general workbench issue and not specific to the editor. I don't know what other applications do in this case, if they resize to fit the new screen content or have some other workarounds to avoid being rendered "under" the keyboard.

**Comment by bpasero** [comment]
@AlanMcBee can you share a screenshot how VSCode is rendered partially below the keyboard?

**Comment by AlanMcBee** [comment]
![image](https://user-images.githubusercontent.com/37203424/72470743-314d6800-3796-11ea-8d57-af976693747b.png)

This is what I see when I follow the repro steps.

Note that the screen might reposition VSCode correctly, once, the first time. When I tried to repro, it looked at first like it was fixed, but it repro'd on the 2nd try.

LMK if this still leaves you with not enough info and I will follow up with more.
